students/Jaidjen/lesson03/Mailroom1.py

```diff
@@ -1,53 +1,60 @@
-import sys  # imports go at the top of the file
+import sys 
 
 
-Mail  =[('James Smith',[2000.25, 500, 27.50]),
+donors  =[('James Smith',[2000.25, 500, 27.50]),
         ('Dorothy Parker',[3175, 475.45, 92.30]),
         ('Dohgyu Hwang',[10670, 350.19,175]),
         ('Sally Yu', [100, 925]),
         ('Jenny Ramos',[25.14])]
 
 prompt = "\n".join(("Functions:",
           "Please choose from below options:",
-          "1 - Send a Thank You email",
+          "1 - Send a Thank You to donors",
           "2 - Create a Report",
           "3 - Quit",
           ">>> "))
 
 
-def view_mail(Mail):
+def add_donors():
+    print("If you want a list of donors please type 'list' ")
+    new_donors = input("Please enter the full name of the donor or select list?  ")
+    for i in range(len(donors)):
+        while True:
+            if new_donors == "list":
+                print("{:<20s}".format(donors[i][0]))
+                break
+            else:
+                donation_amt = int(input("Please enter the donation amount: "))
+                for donor in donors:
+                    if donor[0] == new_donors:
+                        donors.append()
+                        break
+                    else:
+                        donors.append(new_donors)
+                        donors.append(donation_amt)
+                        print('Dear', new_donors, ',')
+                        print('Thank you', new_donors, 'for donating', '{:.2f}'.format(int(donation_amt)))
+                        print('We appreciate your generosity.')
+                        print()
+                        print('Sincerely, Donation Team')
+                        print()
+                        main()
+
+def view_donors(donors):
     print("{:20s}{:20s}{:20s}{:20s}".format("DONOR ", "| TOTAL GIVEN ", "| NUMBER OF GIFTS ", "| AVERAGE GIFTS "))
     print("-----------------------------------------------------------------------------")
 
-    for i in range(len(Mail)):
-        donate=[]
-        dnr_ave= sum(Mail[i][1])
-        ave_amt= int(len(Mail[i][1]))
-        tot_ave= dnr_ave/ave_amt
-        donate.append(dnr_ave)
-        sorted(donate)
-        print("{:20s}{:^20.2f}{:^19d}{:^18.2f}".format(Mail[i][0], dnr_ave, len(Mail[i][1]), tot_ave))
+    summary_data = []
+
+    for donor in donors:
+        donations = donor[1]
+        summary_data.append([((len(donations))),sum(donations)])
+        sorted_data = sorted(summary_data, reverse=True)
```

Comment:
this line needs to be de-dented, meaning moved out of this for loop. there is no need to sort on each loop iteration, you should sort after you're done looping.